articles/qos_deadline_liveliness_lifespan.md

```diff
@@ -0,0 +1,188 @@
+
+This leaves a lot of QoS settings that can only be set if DDS vendor can load additional default settings via a configuration file.
+If a user wants to hook their code into these additional QoS settings then they would need to get a reference to the rmw implementation and program against the vendor specific API.
+Without the abstraction layer provided by ROS their code becomes less portable.
+
+## Proposed Changes
+
+As users start to build more robust applications, they are going to need more control over when data is being delivered and information about when the system is failing.
+To address these needs it was proposed that we start by adding support for the following new DDS QoS settings.
+
+### Proposed New QoS Settings
+
+- Deadline - The deadline policy establishes a contract for how often publishers will receive data and how often subscribers expect to receive data.
+- Liveliness - The liveliness policy is useful for determining if entities on the network are "alive" and notifying interested parties when they are not.
+- Lifespan - The lifespan policy is useful for preventing the delivery of stale messages by specifying a time at which they should no longer be delivered.
+
+More detailed information on these policies can be found below in Appendix A.
+
+### DDS Vendor Support
+
+Some of the primary DDS vendors that ROS supports already provide support for these QoS settings.
+However, the default vendor, Fast-RTPS, does not yet provide full support for these new QoS settings in their library.
+This does not have to be a blocker for adding support for these policy settings to ROS.
+We can start adding support for them and just log warning messages when a user attempts to specify an unsupported QoS setting.
+When support is eventually added for the policies in the vendor library then we can enable it in the rmw implementation.
+
+### Native ROS Support vs Vendor Configuration
+
+It may be argued that many DDS QoS settings could be used with ROS by simply specifying them using the vendor specific configuration mechanism, such as the xml configuration files used by RTI Connext.
+This may be acceptable for some QoS policies that only impact the behavior of the underlying DDS middleware, but in the case where support for QoS policy impacts the behavior of the application code this is not good enough.
+
+For example, it doesn't do much good to specify deadline policy via the vendor configuration mechanism if your ROS application is never notified when a publisher violates the deadline.
+In such cases the application needs to be able to respond to these events in order to log failures, notify users, or adjust system behavior.
```

Comment:
You need to separate specification of the setting from application response to failures related to a setting. Conflating the two concepts is not correct.

Reply:
This concern came up else where as well. I'll be updating the document so it doesn't make a distinction about which policies should or should not be supported natively in the ROS interface based on if their behavior is surfaced in the interface above the rmw layer. 

The next version of the document will attempt to define the policies more in terms of ROS behavior than relying on the assumption of the underlying rmw mechanisms.